clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -433,8 +438,11 @@ private void updateAllValueSchemaEntriesAndLatestValueSchemaEntry(boolean forceR
     }
   }
 
-  private SchemaEntry maybeFetchLatestValueSchemaEntry() {
+  private SchemaEntry maybeFetchLatestValueSchemaEntry(boolean refresh) {
     SchemaEntry latest = latestValueSchemaEntry.get();
+    if (refresh) {
+      shouldRefreshLatestValueSchemaEntry.set(true);
```

Comment:
Probably won't make a difference in the final result but just to be safe and for future-proof can we move this override of `shouldRefreshLatestValueSchemaEntry` inside the `synchronized` block? Since it's possible that concurrently there could be other triggers such as getting schema for an unknown writer schema id that will trigger refresh and set `shouldRefreshLatestValueSchemaEntry` to false after this line.

We can achieve this by entering the critical section and then set
```
if (latest == null || shouldRefreshLatestValueSchemaEntry.get() || refresh) {
  if (refresh) {
    shouldRefreshLatestValueSchemaEntry.set(true);
  }
  ...
}
```

Reply:
Sure, this will be hardened against concurrent calls against schema fetches.